sklearn/utils/sparsefuncs.py

```diff
@@ -25,6 +26,53 @@ def _raise_error_wrong_axis(axis):
         raise ValueError(
             "Unknown axis value: %d. Use 0 for rows, or 1 for columns" % axis)
 
+def csr_csc_min_axis0(X):
+    """Compute the minimum along axis 0 for a csr or csc matrix.
+
+    Parameters
+    ----------
+    X: {csr_matrix, csc_matrix}, shape (n_samples, n_features)
+        Input data.
+
+    Returns
+    -------
+    minimum : array, shape [n_features]
+        The minimum of elements along axis 0.
+    """
+
+    # TODO: I don't know how to avoid this call to `astype()`;
+    # my test cases don't pass with integers otherwise.
+    X = X.astype(float)
+    if isinstance(X, sp.csr_matrix):
+        return csr_min_axis0(X)
+    elif isinstance(X, sp.csc_matrix):
+        return csc_min_axis0(X)
+    else:
+        raise ValueError("Input array was neither a csr_matrix nor csc_matrix.")
+
+def csr_csc_max_axis0(X):
+    """Compute the maximum along axis 0 for a csr or csc matrix.
+
+    Parameters
+    ----------
+    X: {csr_matrix, csc_matrix}, shape (n_samples, n_features)
+        Input data.
+
+    Returns
+    -------
+    maximum : array, shape [n_features]
+        The maximum of elements along axis 0.
+    """
+
+    # TODO: I don't know how to avoid this call to `astype()`;
+    # my test cases don't pass with integers otherwise.
+    X = X.astype(float)
```

Comment:
Is there any way to avoid calling this? I actually run into the following error if I don't convert my data into floats:

    ValueError: Buffer dtype mismatch, expected 'DOUBLE' but got 'long long'

I usually trigger this error on line <a href="https://github.com/scikit-learn/scikit-learn/pull/5825/files#diff-b538b51a4a930c310a7fea81b377e9bbR57">57</a> of `sparsefuncs_fast.pyx`.